core/templates/pages/exploration-editor-page/editor-tab/templates/modal-templates/customize-interaction-modal.component.ts

```diff
@@ -0,0 +1,465 @@
+import { PencilCodeEditorValidationService } from 'interactions/PencilCodeEditor/directives/pencil-code-editor-validation.service';
+import { TextInputValidationService } from 'interactions/TextInput/directives/text-input-validation.service';
+import { InteractiveMapValidationService } from 'interactions/InteractiveMap/directives/interactive-map-validation.service';
+import { MusicNotesInputValidationService } from 'interactions/MusicNotesInput/directives/music-notes-input-validation.service';
+import { FractionInputValidationService } from 'interactions/FractionInput/directives/fraction-input-validation.service';
+import { RatioExpressionInputValidationService } from 'interactions/RatioExpressionInput/directives/ratio-expression-input-validation.service';
+import { Warning } from 'interactions/base-interaction-validation.service';
+import cloneDeep from 'lodash/cloneDeep';
+
+interface CustomizationArgSpecsInterface {
+  name: string;
+  value: unknown;
+  default_value: unknown;
+}
+
+const INTERACTION_SERVICE_MAPPING = {
+  AlgebraicExpressionInputValidationService:
+    AlgebraicExpressionInputValidationService,
+  CodeReplValidationService: CodeReplValidationService,
+  ContinueValidationService: ContinueValidationService,
+  DragAndDropSortInputValidationService: DragAndDropSortInputValidationService,
+  EndExplorationValidationService: EndExplorationValidationService,
+  FractionInputValidationService: FractionInputValidationService,
+  GraphInputValidationService: GraphInputValidationService,
+  ImageClickInputValidationService: ImageClickInputValidationService,
+  InteractiveMapValidationService: InteractiveMapValidationService,
+  ItemSelectionInputValidationService: ItemSelectionInputValidationService,
+  MathEquationInputValidationService: MathEquationInputValidationService,
+  MultipleChoiceInputValidationService: MultipleChoiceInputValidationService,
+  MusicNotesInputValidationService: MusicNotesInputValidationService,
+  NumberWithUnitsValidationService: NumberWithUnitsValidationService,
+  NumericExpressionInputValidationService:
+    NumericExpressionInputValidationService,
+  NumericInputValidationService: NumericInputValidationService,
+  PencilCodeEditorValidationService: PencilCodeEditorValidationService,
+  RatioExpressionInputValidationService: RatioExpressionInputValidationService,
+  SetInputValidationService: SetInputValidationService,
+  TextInputValidationService: TextInputValidationService,
+};
+
+@Component({
+  selector: 'oppia-customize-interaction',
+  templateUrl: './customize-interaction-modal.component.html'
+})
+export class CustomizeInteractionModalComponent
+  extends ConfirmOrCancelModal implements OnInit, AfterContentChecked {
+  @Output() showMarkAllAudioAsNeedingUpdateModalIfRequired:
+    EventEmitter<string[]> = new EventEmitter();
+
+  customizationArgSpecs: CustomizationArgSpecsInterface[];
+  originalContentIdToContent: object;
```

Comment:
`object` type is too generic, don't use until the value is `{}`.

here and elsewhere.

Reply:
in code at a particular point this is `{}`